### PR TITLE
Ability to create predefined calendar or(and) addressbook for new user

### PR DESCRIPTION
**Pull request by itglob**

### CODE REVIEW

File: DOCUMENTATION.md
Comment (pbiering) [inline DOCUMENTATION.md:1]: for what is the added "Default:"?

File: config
Comment (pbiering) [inline config:1]: from this I can't see how to define the option proper

File: radicale/app/__init__.py
Comment (pbiering) [inline radicale/app/__init__.py:1]: the feature should be disabled by default if the property is not set, how is this covered here?

**Comment by pbiering** [comment]
Good idea, but please check my comments.

File: radicale/app/__init__.py
Comment (itglob) [reply radicale/app/__init__.py:1]: disabled by default
json_str return {} (empty dict) 
radicale/app/__init__.py  - line 275  - empty loop

File: config
Comment (itglob) [reply config:1]: is my example in documentation and config unclear?

File: DOCUMENTATION.md
Comment (itglob) [reply DOCUMENTATION.md:1]: "Default:" copied from "hook".
remove ?

**Comment by itglob** [comment]
disabled by default
json_str return {} (empty dict)
```radicale/app/__init__.py``` - line 275 - empty loop


is my example in documentation and config unclear?

**Comment by pbiering** [comment]
I had no time to test so far, but scheduled

**Comment by pbiering** [comment]
Can you please extend the code that it validates the config option during start? It would be helpful to avoid any broken config sniplet detected only on the "new user" trigger.

**Comment by itglob** [comment]
Which aspect should I check? The json syntax is checked during json.loads in config.py.

**Comment by pbiering** [comment]
> Which aspect should I check? The json syntax is checked during json.loads in config.py.

Best that they are valid collection properties (calendar or address book).

Alternative: hardcode and only add an option for enabling of that feature, then the QA can be done by test script.

**Comment by itglob** [comment]
check_and_sanitize_props called before create_collection in ```app/__init__.py - l277```
i didn't think we need do this in config.

**Comment by pbiering** [comment]
> check_and_sanitize_props called before create_collection in `app/__init__.py - l277` i didn't think we need do this in config.

but this will be triggered on first usage only, and then it breaks....I would expect that "preloaded" config is assured working and not waiting to fail on first usage.

**Pull request merged**